src/main/java/org/openjdk/jextract/impl/OutputFactory.java

```diff
@@ -179,17 +179,18 @@ public Void visitScoped(Declaration.Scoped d, Declaration parent) {
         }
 
         boolean isStructKind = Utils.isStructOrUnion(d);
+        JavaSourceBuilder prevBuilder = null;
```

Comment:
Note that `StructBuilder` no longer keeps a reference to the enclosing builder. So, it can not be returned from `classEnd`/`end`. Instead, we keep track of the enclosing builder here.

This also helps prevent StructBuilder from being too context aware.